tests are failing

## Issue

One of the test is failing when it is executed

```
make docker-build IMG=my-app-operator
go generate ./pkg/... ./cmd/...
go fmt ./pkg/... ./cmd/...
go vet ./pkg/... ./cmd/...
go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go all
CRD manifests generated under '/Users/dabou/Code/go-workspace/src/github.com/kubernetes-sigs/application/config/crds' 
RBAC manifests generated under '/Users/dabou/Code/go-workspace/src/github.com/kubernetes-sigs/application/config/rbac' 
go test ./pkg/... ./cmd/... -coverprofile cover.out
?       github.com/kubernetes-sigs/application/pkg/apis [no test files]
?       github.com/kubernetes-sigs/application/pkg/apis/app     [no test files]
ok      github.com/kubernetes-sigs/application/pkg/apis/app/v1beta1     11.795s coverage: 9.1% of statements
ok      github.com/kubernetes-sigs/application/pkg/component    0.053s  coverage: 40.0% of statements
?       github.com/kubernetes-sigs/application/pkg/controller   [no test files]
2019/02/05 19:25:35 *v1beta1.Application/default/foo Validating spec
2019/02/05 19:25:35 *v1beta1.Application/default/foo Applying defaults
2019/02/05 19:25:35 *v1beta1.Application/default/foo(cmpnt:app)  { reconciling component
2019/02/05 19:25:35 *v1beta1.Application/default/foo(cmpnt:app)  Expected Resources:
2019/02/05 19:25:35 *v1beta1.Application/default/foo(cmpnt:app)  Observed Resources:
2019/02/05 19:25:35 *v1beta1.Application/default/foo(cmpnt:app)  Reconciling Resources:
2019/02/05 19:25:35 *v1beta1.Application/default/foo(cmpnt:app)  } reconciling component
2019/02/05 19:25:35 *v1beta1.Application/default/foo Validating spec
2019/02/05 19:25:35 *v1beta1.Application/default/foo Applying defaults
2019/02/05 19:25:35 *v1beta1.Application/default/foo(cmpnt:app)  { reconciling component
2019/02/05 19:25:35 *v1beta1.Application/default/foo(cmpnt:app)  Expected Resources:
2019/02/05 19:25:35 *v1beta1.Application/default/foo(cmpnt:app)  Observed Resources:
2019/02/05 19:25:35 *v1beta1.Application/default/foo(cmpnt:app)  Reconciling Resources:
2019/02/05 19:25:35 *v1beta1.Application/default/foo(cmpnt:app)  } reconciling component
--- FAIL: TestReconcile (5.23s)
    application_controller_test.go:77: 
        Timed out after 5.004s.
        Expected success, but got an error:
            <*errors.StatusError | 0xc0002b55f0>: {
                ErrStatus: {
                    TypeMeta: {Kind: "", APIVersion: ""},
                    ListMeta: {SelfLink: "", ResourceVersion: "", Continue: ""},
                    Status: "Failure",
                    Message: "Deployment.apps \"foo-deployment\" not found",
                    Reason: "NotFound",
                    Details: {
                        Name: "foo-deployment",
                        Group: "apps",
                        Kind: "Deployment",
                        UID: "",
                        Causes: nil,
                        RetryAfterSeconds: 0,
                    },
                    Code: 404,
                },
            }
            Deployment.apps "foo-deployment" not found
FAIL
coverage: 75.0% of statements
FAIL    github.com/kubernetes-sigs/application/pkg/controller/application       16.850s
?       github.com/kubernetes-sigs/application/pkg/customresource       [no test files]
ok      github.com/kubernetes-sigs/application/pkg/finalizer    0.043s  coverage: 94.7% of statements
?       github.com/kubernetes-sigs/application/pkg/genericreconciler    [no test files]
ok      github.com/kubernetes-sigs/application/pkg/kbcontroller 0.031s  coverage: 0.0% of statements
ok      github.com/kubernetes-sigs/application/pkg/resource     0.063s  coverage: 27.8% of statements
?       github.com/kubernetes-sigs/application/cmd/manager      [no test files]
make: *** [test] Error 1
```

--- a/pkg/controller/application/application_controller_test.go
+++ b/pkg/controller/application/application_controller_test.go
@@ -24,7 +24,6 @@ import (
 	kbc "github.com/kubernetes-sigs/application/pkg/kbcontroller"
 	"github.com/onsi/gomega"
 	"golang.org/x/net/context"
-	appsv1 "k8s.io/api/apps/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
@@ -42,7 +41,7 @@ const timeout = time.Second * 5
 
 func TestReconcile(t *testing.T) {
 	g := gomega.NewGomegaWithT(t)
-	instance := &appv1beta1.Application{ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "default"}}
+	instance := &appv1beta1.Application{ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "default"}, Spec: appv1beta1.ApplicationSpec{}}
 
 	// Setup the Manager and Controller.  Wrap the Controller Reconcile function so it writes each request to a
 	// channel when it is finished.
@@ -71,18 +70,5 @@ func TestReconcile(t *testing.T) {
 	g.Expect(err).NotTo(gomega.HaveOccurred())
 	defer c.Delete(context.TODO(), instance)
 	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedRequest)))
-
-	deploy := &appsv1.Deployment{}
-	g.Eventually(func() error { return c.Get(context.TODO(), depKey, deploy) }, timeout).
-		Should(gomega.Succeed())
-
-	// Delete the Deployment and expect Reconcile to be called for Deployment deletion
-	g.Expect(c.Delete(context.TODO(), deploy)).NotTo(gomega.HaveOccurred())
-	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedRequest)))
-	g.Eventually(func() error { return c.Get(context.TODO(), depKey, deploy) }, timeout).
-		Should(gomega.Succeed())
-
-	// Manually delete Deployment since GC isn't enabled in the test control plane
-	g.Expect(c.Delete(context.TODO(), deploy)).To(gomega.Succeed())
 
 }

